src/hooks/useWebRTC.ts

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useRef, useState } from "react";
+import { Signal, useSignal } from "./useSignal";
+
+export type WebRTCState = "rejected" | RTCPeerConnectionState
+
+export function useWebRTC(signalAddr: string, config?: RTCConfiguration) {
+    const videoElement = useRef<HTMLVideoElement>(null!)
+    const [state, setState] = useState<WebRTCState>("new")
+    const remotePeer = useRef<RTCPeerConnection>(null!)
+
+    function onSignal(signal: Signal) {
+        if (signal.event === "answer") {
```

Comment:
Put the string values ​​in a map above the document and query the map with a get in the function. This is not a good practice